plugins/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/condition/NamedResourceExist.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v 1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.condition;
+
+import static org.junit.Assert.assertNotNull;
+
+import org.jboss.reddeer.common.condition.AbstractWaitCondition;
+import org.jboss.tools.openshift.core.connection.Connection;
+
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.ResourceKind;
+
+/**
+ * Wait condition to wait for existence of a (core, openshift client-lib)
+ * resource of a given kind and name in the given project.
+ * 
+ * @author adietish@redhat.com
+ *
+ */
+public class NamedResourceExist extends AbstractWaitCondition {
+
+	private final String kind;
+	protected final String name;
+	private String project;
+	protected Connection connection;
+	
+	public NamedResourceExist(String name, Connection connection) {
+		this(ResourceKind.PROJECT, name, null, connection);
+	}
+
+	public NamedResourceExist(String kind, String name, String project, Connection connection) {
```

Comment:
Add javadoc. Its not clear enough what the kind string should be. Add reference to ResourceKind in the javadoc.